packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -49,250 +49,306 @@ const password = 'password123';
 const commonConfig = { chain: 'goerli', hardfork: 'berlin' };
 
 describe('KeyringController', () => {
-  let keyringController: KeyringController;
-  let preferences: {
-    setAccountLabel: sinon.SinonStub;
-    removeIdentity: sinon.SinonStub;
-    syncIdentities: sinon.SinonStub;
-    updateIdentities: sinon.SinonStub;
-    setSelectedAddress: sinon.SinonStub;
-  };
-  let initialState: {
-    isUnlocked: boolean;
-    keyringTypes: string[];
-    keyrings: Keyring[];
-  };
-  const encryptor = new MockEncryptor();
-  const additionalKeyrings = [QRKeyring];
-  const additionalKeyringBuilders = additionalKeyrings.map((keyringType) =>
-    keyringBuilderFactory(keyringType),
-  );
-  const baseConfig: Partial<KeyringConfig> = {
-    encryptor,
-    keyringBuilders: additionalKeyringBuilders,
-  };
-
-  beforeEach(async () => {
-    preferences = {
-      setAccountLabel: sinon.stub(),
-      removeIdentity: sinon.stub(),
-      syncIdentities: sinon.stub(),
-      updateIdentities: sinon.stub(),
-      setSelectedAddress: sinon.stub(),
-    };
-    keyringController = new KeyringController(preferences, baseConfig);
-
-    initialState = await keyringController.createNewVaultAndKeychain(password);
-  });
-
   afterEach(() => {
     sinon.restore();
   });
 
-  it('should set default state', () => {
-    expect(keyringController.state.keyrings).not.toStrictEqual([]);
-    const keyring = keyringController.state.keyrings[0];
-    expect(keyring.accounts).not.toStrictEqual([]);
-    expect(keyring.index).toStrictEqual(0);
-    expect(keyring.type).toStrictEqual('HD Key Tree');
+  it('should set default state', async () => {
```

Comment:
To fit the shape of the other tests in this file, what do you think about placing this in a `constructor` describe block? Like:

``` typescript
describe('constructor', () => {
  await withController({}, async ({ controller }) => {
    expect(controller.state.keyrings).not.toStrictEqual([]);
    const keyring = controller.state.keyrings[0];
    expect(keyring.accounts).not.toStrictEqual([]);
    expect(keyring.index).toStrictEqual(0);
    expect(keyring.type).toStrictEqual('HD Key Tree');
  });
});
```

Reply:
Good call, but actually here we don't test the result of the constructor, but the result of calling the constructor _and_ `createNewVaultAndKeychain` right after it. So might make sense to move those there

Reply:
Right, makes sense.